### DISCUSSION TITLE
RAID-Z2 on 7 disks, but one disk is bigger than rest

### DISCUSSION THREAD

**Discussion by Neptunet-GitHub** [openzfs/zfs#16775] [Q&A] [answered]
Hi guys,

Maybe someone can explain behaviour. 

I tested ZFS what done if I created RAID-Zn on disks but with different size. For example: 6 Drives 10 GB + 1 Drive 15 GB. In this case, I expected, that RAID-Z2 would limit all drives to the smallest size (10GB), effectively reducing the 15 GB drive to 10 GB and create zpool on 7 disks (10 GB each) - 2 disks (parity) = 50GB. 

But I got:
![image](https://github.com/user-attachments/assets/654513ba-c00b-46f5-b5a4-0a06e70ea172)
available size from ZFS:
![image](https://github.com/user-attachments/assets/1bf7cdfd-6700-4cf1-bf39-b01e088a8f8b)
and available size from linux perspective:
![image](https://github.com/user-attachments/assets/7e320e3c-f5f4-4b45-bca2-b68c7ace8a11)

Can someone explain if it's correct and why?

openZFS version 2.2.7

**Comment by amotin** [comment, marked as answer]
ZFS prints capacities in binary powers, i.e. GibiBytes here, while disk vendors use decimal GigaBytes.  So 50GB is indeed about 46GiB. And the difference likely belong to ZFS reserve.  The 15GB disk indeed got truncated.

**Reply by Neptunet-GitHub** [reply]
Thanks @amotin - now all is clear :) - It's good, that everything is working as it should.
